integration_tests/features/support/steps.js

```diff
@@ -965,24 +1008,103 @@ Then(/the number of coinbase transactions for wallet (.*) and wallet (.*) are (.
             walletStats[0][1] + walletStats[1][1]
         );
     } else {
-        except("\nCoinbase comparison: Not enough results saved on the stack!").to.equal("")
+        expect("\nCoinbase comparison: Not enough results saved on the stack!").to.equal("")
     }
 });
 
+Then(/all coinbase transactions for wallet (.*) and wallet (.*) have consistent but opposing validity/, {timeout: 20*1000}, async function (
+    walletNameA,
+    walletNameB
+) {
+    let walletClientA = this.getWallet(walletNameA).getClient();
+    let transactionsA = await walletClientA.getAllCoinbaseTransactions();
+    let walletClientB = this.getWallet(walletNameB).getClient();
+    let transactionsB = await walletClientB.getAllCoinbaseTransactions();
+    if ((transactionsA === undefined) || (transactionsB === undefined)) {
+        expect("\nNo transactions found!").to.equal("")
+    }
+    let validA = transactionsA[0]["valid"];
+    for (let i = 0; i<transactionsA.length; i++) {
+        if (transactionsA[0]["valid"] != transactionsA[i]["valid"]) {
+            expect("\n" + walletNameA + "'s coinbase transactions do not have a consistent validity status").to.equal("")
+        };
+    }
+    let validB = transactionsB[0]["valid"];
+    for (let i = 0; i<transactionsB.length; i++) {
+        if (transactionsB[0]["valid"] != transactionsB[i]["valid"]) {
+            expect("\n" + walletNameB + "'s coinbase transactions do not have a consistent validity status").to.equal("")
+        };
+    }
+    expect(validA).to.equal(!validB)
+});
+
+Then(/all normal transactions for wallet (.*) and wallet (.*) have consistent but opposing validity/, {timeout: 20*1000}, async function (
```

Comment:
Can't these two methods be combined with a regular expression rather? There is a lot of code being duplicated here

Reply:
Yip, will combine